### PR TITLE
[PLAT-3702] Report runtime versions for Delayed Job, Mailman and Shoryuken

### PR DESCRIPTION
**Pull request by imjoehaines**
## Goal

This PR adds runtime versions for:

- Delayed Job
- Mailman
- Shoryuken

Mailman & Shoryuken are straightforward, but Delayed Job doesn't have a constant with its version in so instead we have to read it from the loaded Gems

## Review

For the submitter, initial self-review:

- [x] Commented on code changes inline explain the reasoning behind the approach
- [x] Reviewed the test cases added for completeness and possible points for discussion
- [x] A changelog entry was added for the goal of this pull request
- [x] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [x] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: lib/bugsnag/integrations/delayed_job.rb [hunk 1]
```diff
@@ -10,11 +10,22 @@
 module Delayed
   module Plugins
     class Bugsnag < ::Delayed::Plugin
+      ##
+      # DelayedJob doesn't have an easy way to fetch its version, but we can use
+      # Gem.loaded_specs to get the version instead
+      def self.delayed_job_version
+        ::Gem.loaded_specs['delayed_job'].version.to_s
+      rescue StandardError
+        nil
```
Comment (imjoehaines) [inline lib/bugsnag/integrations/delayed_job.rb:19]: This is to make Rubocop happy — if we don't explicitly put a `nil` here it gets upset about suppressing exceptions

Comment (kattrali) [reply lib/bugsnag/integrations/delayed_job.rb:19]: This should get a little comment or something - I could easily see deleting the `nil` in the future because it doesn't normally matter.
